File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼.java
class Solution {
    public int factorial(int num) {
        int result = 1;
        for (int i = 2; i <= num ; i++) {
            result *= i;
        }
        return result;
    }
    
    public int solution(int n) {
        int num = 1;

        while(true) {
            int temp = factorial(num);
            if(temp > n) {
                num--;
                break;
            }
            if(temp == n) {
                break;
            }
            num++;
        }
        return num;
    }
}

File: C/프로그래머스/0/120848. 팩토리얼/팩토리얼.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int factorial(int num) {
    int tmp = 1;
    for(int i = 1; i <= num; i++) {
        tmp *= i;
    }
    return tmp;
}
int solution(int n) {
    int answer = 0;
    for(int i = 1; i <= 10; i++) {
        int temp = factorial(i);
        if(temp > n) {
            answer = i - 1;
            break;
        } else if (temp == n) {
            answer = i;
            break;
        }
    }
    return answer;
}